src/averageAges.js

```diff
@@ -37,9 +45,23 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(person => person.sex === 'f');
+
+  if (withChildren === true) {
+    womens = womens.filter(woman => people.some(person =>
+      person.mother === woman.name));
```

Comment:
The 'withChildren' check is correct, but you might simplify it for better readability: `womens = womens.filter(woman => people.some(({ mother: person }) => person === woman.name));`